spec/unit/rust-crypto/verification.spec.ts

```diff
@@ -152,3 +405,148 @@ function makeMockedInner(): Mocked<RustSdkCryptoJs.VerificationRequest> {
+                resp = {
+                    device_keys: {},
+                };
+                const body = JSON.parse(request.body);
+                const query = body.device_keys;
+                const masterKeys: Record<string, any> = {};
+                const selfSigningKeys: Record<string, any> = {};
+                if (ourUserId in query) {
+                    resp.device_keys[ourUserId] = { [ourDeviceId]: ourDeviceKeys };
+                    masterKeys[ourUserId] = ourCrossSigningKeys.master_key;
+                    selfSigningKeys[ourUserId] = ourCrossSigningKeys.self_signing_key;
+                    resp.user_signing_keys = {
+                        [ourUserId]: ourCrossSigningKeys.user_signing_key,
+                    };
+                }
+                if (theirUserId in query) {
+                    resp.device_keys[theirUserId] = {
+                        [theirDeviceId]: theirDeviceKeys,
+                    };
+                    masterKeys[theirUserId] = theirCrossSigningKeys.master_key;
+                    selfSigningKeys[theirUserId] = theirCrossSigningKeys.self_signing_key;
+                }
+                if (Object.keys(masterKeys).length) {
+                    resp.master_keys = masterKeys;
+                }
+                if (Object.keys(selfSigningKeys).length) {
+                    resp.self_signing_keys = selfSigningKeys;
+                }
+            } else if (request instanceof RustSdkCryptoJs.RoomMessageRequest) {
+                resp = { event_id: "$" + request.event_type };
+                theirOlmMachine.receiveVerificationEvent(
+                    JSON.stringify({
+                        type: request.event_type,
+                        sender: ourUserId,
+                        event_id: "$" + request.event_type,
+                        content: JSON.parse(request.body),
+                        origin_server_ts: Date.now(),
+                        unsigned: {
+                            age: 0,
+                        },
+                    }),
+                    new RustSdkCryptoJs.RoomId(request.room_id),
+                );
+            } else if (request instanceof RustSdkCryptoJs.SignatureUploadRequest) {
+                // this only gets called at the end after the verification
+                // succeeds, so we don't actually have to do anything.
+                resp = { failures: {} };
+            }
+        }
+        if (!(request instanceof RustSdkCryptoJs.UploadSigningKeysRequest) && request.id) {
+            await ourOlmMachine.markRequestAsSent(request.id!, request.type, JSON.stringify(resp));
+        }
+    }
+
+    const promise = (async () => {
+        while (!stopRequestLoop) {
+            const requests = await ourOlmMachine.outgoingRequests();
+            for (const request of requests) {
+                await makeOutgoingRequest(request);
+            }
+        }
+    })();
```

Comment:
this is a bit hard to read.

```suggestion
    async function runLoop() {
        while (!stopRequestLoop) {
            const requests = await ourOlmMachine.outgoingRequests();
            for (const request of requests) {
                await makeOutgoingRequest(request);
            }
        }
    }
    
    const loopCompletedPromise = runLoop();
```